zebra-state/src/service/finalized_state/zebra_db/block.rs

```diff
@@ -79,6 +81,38 @@ impl ZebraDb {
         self.db.zs_get(block_by_height, &height)
     }
 
+    /// Returns the Sapling
+    /// [`NoteCommitmentTree`](sapling::tree::NoteCommitmentTree) specified by a
+    /// hash or height, if it exists in the finalized `db`.
+    pub fn sapling_tree(
+        &self,
+        hash_or_height: HashOrHeight,
+    ) -> Option<Arc<sapling::tree::NoteCommitmentTree>> {
+        let height_by_hash = self.db.cf_handle("height_by_hash").unwrap();
+        let height = hash_or_height.height_or_else(|hash| self.db.zs_get(height_by_hash, &hash))?;
+
+        let sapling_note_commitment_tree_handle =
+            self.db.cf_handle("sapling_note_commitment_tree").unwrap();
+
+        self.db.zs_get(sapling_note_commitment_tree_handle, &height)
+    }
+
+    /// Returns the Orchard
+    /// [`NoteCommitmentTree`](orchard::tree::NoteCommitmentTree) specified by a
+    /// hash or height, if it exists in the finalized `db`.
+    pub fn orchard_tree(
+        &self,
+        hash_or_height: HashOrHeight,
+    ) -> Option<Arc<orchard::tree::NoteCommitmentTree>> {
+        let height_by_hash = self.db.cf_handle("height_by_hash").unwrap();
+        let height = hash_or_height.height_or_else(|hash| self.db.zs_get(height_by_hash, &hash))?;
```

Comment:
Optional simplification:

```suggestion
        let height = hash_or_height.height_or_else(|hash| self.height(hash))?;
```